ui/lib/components/TimeRangeSelector/index.tsx

```diff
@@ -1,60 +1,69 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useMemo } from 'react'
 import { Dropdown, Button, DatePicker } from 'antd'
 import { ClockCircleOutlined, DownOutlined } from '@ant-design/icons'
 import { getValueFormat } from '@baurine/grafana-value-formats'
 import cx from 'classnames'
 import dayjs from 'dayjs'
+import moment, { Moment } from 'moment'
 import { useTranslation } from 'react-i18next'
 
-import styles from './TimeRangeSelector.module.less'
-import { Moment } from 'moment'
+import styles from './index.module.less'
 
 const { RangePicker } = DatePicker
 
 const RECENT_MINS = [30, 60, 3 * 60, 6 * 60, 12 * 60, 24 * 60]
 
 export interface TimeRange {
-  recent: number
+  recent: number // 0 means absolute time range, else means relative time range
   begin_time: number
   end_time: number
 }
 
-export function getDefTimeRange(): TimeRange {
+function calcTimeRange(mins: number): TimeRange {
   const now = dayjs().unix()
   return {
-    recent: 30,
-    begin_time: now - 30 * 60,
+    recent: mins,
+    begin_time: now - mins * 60,
     end_time: now,
   }
 }
 
+export function getDefTimeRange(): TimeRange {
```

Comment:
Looks like normally components do not provide this kind of function. How about using `value == null` to denote a default selection?